Guide/src/dev_guide/dev_tools/xflowey.md

```diff
@@ -10,13 +10,615 @@ for writing maintainable, cross-platform automation.
+5. [Variables: ReadVar and WriteVar](#variables-readvar-and-writevar)
+6. [Emitting Steps](#emitting-steps)
+7. [Runtime Services](#runtime-services)
+8. [Node Design Philosophy](#node-design-philosophy)
+9. [Common Patterns](#common-patterns)
+
+---
+
+## Core Concepts
+
+### Two-Phase Execution Model
+
+Flowey operates in two distinct phases:
+
+1. **Build-Time (Resolution Phase)**: When you run `cargo xflowey regen`, flowey:
+   - Reads `.flowey.toml` to determine which pipelines to regenerate
+   - Builds the flowey binary (e.g., `flowey-hvlite`) via `cargo build`
+   - Runs the flowey binary with `pipeline <backend> --out <file> <cmd>` for each pipeline definition
+   - During this invocation, flowey constructs a directed acyclic graph (DAG) by:
+     - Instantiating all nodes defined in the pipeline
+     - Processing their requests
+     - Resolving dependencies between nodes via variables and artifacts
+     - Determining the execution order
+     - Performing flowey-specific validations (dependency resolution, type checking, etc.)
+   - Generates YAML files for CI systems (ADO, GitHub Actions) at the paths specified in `.flowey.toml`
+
+2. **Runtime (Execution Phase)**: The generated YAML is executed by the CI system (or locally via `cargo xflowey <pipeline>`). Steps run in the order determined at build-time:
+   - Variables are read and written with actual values
+   - Commands are executed
+   - Artifacts are published/consumed
+   - Side effects occur
+
+```admonish note
+**Understanding the Workflow:**
+
+The `.flowey.toml` file at the repo root defines which pipelines to generate and where. For example:
+```toml
+[[pipeline.flowey_hvlite.github]]
+file = ".github/workflows/openvmm-pr.yaml"
+cmd = ["ci", "checkin-gates", "--config=pr"]
+```
+
+When you run `cargo xflowey regen`:
+1. It reads `.flowey.toml` 
+2. Builds the `flowey-hvlite` binary
+3. Runs `flowey-hvlite pipeline github --out .github/workflows/openvmm-pr.yaml ci checkin-gates --config=pr`
+4. This generates/updates the YAML file with the resolved pipeline
+
+**Key Distinction:**
+- `cargo build -p flowey-hvlite` - Only compiles the flowey code to verify it builds successfully. **Does not** construct the DAG or generate YAML files.
+- `cargo xflowey regen` - Compiles the code **and** runs the full build-time resolution to construct the DAG, validate the pipeline, and regenerate all YAML files defined in `.flowey.toml`.
+
+Always run `cargo xflowey regen` after modifying pipeline definitions to ensure the generated YAML files reflect your changes.
+```
+
+This separation allows flowey to:
+- Validate the entire workflow before execution
+- Generate static YAML for CI systems (ADO, GitHub Actions)
+- Optimize step ordering and parallelization
+- Catch dependency errors at build-time rather than runtime
+
+### Backend Abstraction
+
+Flowey supports multiple execution backends:
+
+- **Local**: Runs directly on your development machine via bash or direct
```

Comment:
there is no "bash" backend. flowey's whole schtick is that it lets you avoid writing bash entirely, and instead write those flue commands in Rust itself (using constructs like xshell, which under the hood, is just invoking std::process stuff - no bash in the loop at all). this is also essential to making flowey pipelines ostensibly cross-platform between windows and linux (since using bash on windows is a road to much pain)